R/cluster.R

```diff
@@ -0,0 +1,150 @@
+#' Detect (spatial) groundwater well clusters
+#'
+#' \code{cluster_locs()} accepts as input a
+#' dataframe with X/Y coordinates, or an \code{sf} object
+#' of geometry type \code{POINT}.
+#' The function adds an integer variable that defines cluster membership.
+#' The intention is to detect spatial groundwater well clusters; hence it uses a
+#' sensible method of spatial clustering and default euclidean distance
+#' to cut the cluster tree.
+#'
+#' The function performs agglomerative clustering with the
+#' \emph{complete linkage} method.
+#' This way, the application of a tree cutoff (\code{max_dist}) means that each
+#' cluster is a collection of locations with a maximum distance - between any
+#' two locations of the cluster - not larger than the cutoff value.
+#' All locations that can be clustered under this condition, will be.
+#' Locations that can not be clustered receive a unique cluster value.
+#'
+#' The function's code was partly inspired by unpublished code from Ivy Jansen.
+#'
+#' @param input A dataframe with X/Y coordinates, or an \code{sf} object of
+#' geometry type \code{POINT}.
+#' A typical input dataframe is the collected output of \code{\link{get_locs}}.
+#' @param max_dist The maximum geospatial distance between two points to make
+#' them belong to the same cluster.
+#' The default value is sensible when using \emph{meter} as unit of the
```

Comment:
Whether the default is sensible not only depends on the unit of measurement but also on the application at hand?

Reply:
True, will add that. It has been tried (and tested with @w-jan) to cover most common usecases.